internal/scanner/scanner.go

```diff
@@ -367,30 +369,37 @@ func (s *Scanner) produceTests(ctx context.Context, n int) <-chan *testWork {
 	go func() {
 		defer close(testChan)
 
-		var testHeaderValue string
+		var debugHeader string
+
+		hash := sha256.New()
 
 		for _, t := range testCases {
 			for _, payload := range t.Payloads {
 				for _, e := range t.Encoders {
 					for _, placeholder := range t.Placeholders {
-						if s.isTestEnv {
-							testHeaderValue = fmt.Sprintf(
-								"set=%s,name=%s,placeholder=%s,encoder=%s",
-								t.Set, t.Name, placeholder, e,
-							)
+						if s.enableDebugHeader {
+							hash.Write([]byte(t.Set))
+							hash.Write([]byte(t.Name))
+							hash.Write([]byte(placeholder))
+							hash.Write([]byte(e))
+							hash.Write([]byte(payload))
+
+							debugHeader = hex.EncodeToString(hash.Sum(nil))
+
+							hash.Reset()
```

Comment:
Just a question/suggestion: isn't it better to reset the hash before usage, not after the usage? It is the same in terms of code usage, but in terms of human mistakes it is better to clear the data before writing, and only after that write the new data (so we definitely won't miss the reset operation somewhere).

Most of the examples that I saw resets the hash first: https://golang.hotexamples.com/examples/hash/Hash/Reset/golang-hash-reset-method-examples.html

But maybe I am wrong.

Reply:
Yep, it makes sense. In the first implementation I reset the hash before using it. Fixed - https://github.com/wallarm/gotestwaf/pull/160/commits/ff81926b3bbd7ce6749ce79d4b602baa95b7c1c1